cxx-qt-lib/src/update_requester.cpp

```diff
@@ -28,5 +28,21 @@ UpdateRequester::requestUpdate() const
   return QMetaObject::invokeMethod(m_obj, m_method, Qt::QueuedConnection);
 }
 
+QtThread::QtThread(QPointer<QObject> obj, std::function<void()> lambda)
+  : m_lambda(lambda)
+  , m_obj(obj)
+{
+}
+
+bool
+QtThread::queue() const
```

Comment:
this is no safer than the old approach yet, but the API is nicer and makes the parsing/generation easier as we remove the `impl UpdateRequestHandler for T` trait

Reply:
Fixed now have locking :-)